src/main/scala/mesosphere/marathon/core/task/termination/impl/TaskKillProgressActor.scala

```diff
@@ -0,0 +1,70 @@
+package mesosphere.marathon.core.task.termination.impl
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import mesosphere.marathon.KillingTasksFailedException
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.core.event.MesosStatusUpdateEvent
+
+import scala.concurrent.Promise
+import scala.collection.mutable
+import scala.util.Try
+
+/**
+  * This actor watches over a given set of taskIds and completes a promise when
+  * all tasks have been reported terminal (including LOST et al)
+  *
+  * @param taskIds the taskIds that shall be watched.
+  * @param promise the promise that shall be completed when all tasks have been
+  *                reported terminal.
+  */
+class TaskKillProgressActor(taskIds: mutable.HashSet[Task.Id], promise: Promise[Unit]) extends Actor with ActorLogging {
+  // TODO: if one of the watched task is reported terminal before this actor subscribed to the event bus,
+  //       it won't receive that event. should we reconcile tasks after a certain amount of time?
+
+  // this should be used for logging to prevent polluting the logs
+  val name = "TaskKillProgressActor" + self.hashCode()
+
+  override def preStart(): Unit = {
+    if (taskIds.nonEmpty) {
+      context.system.eventStream.subscribe(self, classOf[MesosStatusUpdateEvent])
+      log.info("Starting {} to track kill progress of {} tasks", name, taskIds.size)
+    } else {
+      promise.tryComplete(Try(()))
+      log.info("premature aborting of {} - no tasks to watch for", name)
+      context.stop(self)
+    }
+  }
+
+  override def postStop(): Unit = {
+    context.system.eventStream.unsubscribe(self)
+
+    if (!promise.isCompleted) {
```

Comment:
One possible condition: a new leader gets elected, and not all tasks where killed.
Is there any transient state, that does not get reconciled:
- deployments are reconciled
- healthchecks would be executed again on the new leader
- scale changes would be executed again on the new leader

But
- unsuccessful kills via the API would get lost eventually